extensions/wikia/ContentReview/tests/ContentReviewHelperTest.php

```diff
@@ -98,96 +156,103 @@ public function testHasPageApprovedId( $latestReviewedId, $oldId, $expected, $me
 		$this->assertEquals( $expected, $this->getHelper()->hasPageApprovedId( $modelMock, 0, 0, $oldId ), $message );
 	}
 
+	public function isPageForReviewTestData() {
+		$goodMimeType = 'text/javascript';
+		$badMimeType = 'text/css';
+		return [
+			[
+				false, // isJsPage()
+				false, // inNamespace()
+				$badMimeType,
+				false, // expected
+			],
+			[
+				false, // isJsPage()
+				true, // inNamespace()
+				$badMimeType,
+				false, // expected
+			],
+			[
+				false, // isJsPage()
+				false, // inNamespace()
+				$goodMimeType,
+				false, // expected
+			],
+			[
+				false, // isJsPage()
+				true, // inNamespace()
+				$goodMimeType,
+				true, // expected
+			],
+			[
+				true, // isJsPage()
+				false, // inNamespace()
+				$badMimeType,
+				true, // expected
+			],
+		];
+	}
+
 	public function replaceWithLastApprovedRevisionProvider() {
-		$pageId = 123;
-		$revId1 = 566;
-		$revId2 = 567;
-		$revIdNull = null;
-		$textEmpty = '';
-		$text1 = '';
-		$text2 = '';
-		$jsType = 'jstype';
+		$originalText = 'This is the original text.';
+		$lastReviewedText = 'This is the last reviewed text.';
+		$emptyText = '';
+
 		return [
 			[
-				[
-					'pageId' => $pageId,
-					'latestRevId' => $revId1,
-					'isJsPage' => true,
-					'contentType' => 'no impact in this test, random 6519846169498',
-					'latestApprovedRevData' => [
-						'revision_id' => $revId1
-					],
-					'latestApprovedRevText' => 'revision text',
-					'isContentReviewTestModeEnabled' => false,
-					'text' => $text1,
-				],
-				$text1,
-				'Latest revision id same as latest approved revision',
+				false, // isPageReviewed()
+				0, // getLatestRevID()
+				[], // getLatestReviewedRevision()
+				false, // isContentReviewTestModeEnabled()
+				false, // $revisionExists
+				$originalText,
+				$lastReviewedText,
+				$originalText, // $expectedText
+				$message = 'Case 1 - a page does not qualify to be reviewed (isPageReviewed === false).',
 			],
 			[
-				[
-					'pageId' => $pageId,
-					'latestRevId' => $revId2,
-					'isJsPage' => true,
-					'contentType' => 'no impact in this test, random 4563786783453',
-					'latestApprovedRevData' => [
-						'revision_id' => $revId1
-					],
-					'latestApprovedRevText' => $text1,
-					'isContentReviewTestModeEnabled' => false,
-					'text' => $text2,
-				],
-				$text1,
-				'Current text replaced with last approved revision',
+				true, // isPageReviewed()
+				100, // getLatestRevID()
+				[ 'revision_id' => 100 ], // getLatestReviewedRevision()
+				false, // isContentReviewTestModeEnabled()
+				false, // $revisionExists
```

Comment:
The value does not really matter, since the script should not get to it, but it makes more sense in terms of crafting a scenario. Thanks! :)